public/js/ace-1.2.5/mode-latex.js

```diff
@@ -205,47 +205,116 @@ var LatexFoldMode = require("./folding/latex").FoldMode;
 var Range = require("../range").Range;
 var WorkerClient = require("ace/worker/worker_client").WorkerClient;
 
-   
+
 var Mode = function() {
     this.HighlightRules = LatexHighlightRules;
     this.foldingRules = new LatexFoldMode();
     this.createWorker = function(session) {
-	var worker = new WorkerClient(["ace"], "ace/mode/latex_worker", "LatexWorker");
+	var doc = session.getDocument();
+	var selection = session.getSelection();
+
 	var savedRange = {};
+	var suppressions = [];
+	var hints = [];
+	var changeHandler = null;
 
-	worker.attachToDocument(session.getDocument());
+	var worker = new WorkerClient(["ace"], "ace/mode/latex_worker", "LatexWorker");
+	worker.attachToDocument(doc);
 
-	worker.on("lint", function(results) {
+	doc.on("change", function () {
+	    if(changeHandler) {
+		clearTimeout(changeHandler);
+		changeHandler = null;
+	    }
+	});
+
+	selection.on("changeCursor", function () {
+	    if(suppressions.length > 0) {
+		changeHandler = setTimeout(function () {
+		    updateMarkers();
+		    suppressions = [];
+		    changeHandler = null;
+		}, 100);
+	    }
+	});
+
+	var updateMarkers = function () {
+	    var annotations = [];
 	    var newRange = {};
-	    for (var i = 0; i<results.data.length; i++) {
-		var start_row = results.data[i].start_row;
-		var end_row = results.data[i].end_row;
-		var key = start_row + ":" + end_row;
-		newRange[key] = results.data[i];
+	    var cursor = selection.getCursor();
+	    suppressions = [];
+
+	    for (var i = 0; i<hints.length; i++) {
+		var data = hints[i];
+		var start_row = data.start_row;
+		var start_col = data.start_col;
+		var end_row = data.end_row;
+		var end_col = data.end_col;
+		if (data.suppressIfEditing &&
+		    ((cursor.row === start_row && cursor.column == start_col+1)
```

Comment:
bit of a confusing if statment, could be broken down into variables above then done as `if (a || b || c || d)` so it is clear what is being checked